src/System.Net.Sockets/src/System.Net.Sockets.csproj

```diff
@@ -223,9 +213,178 @@
+      <Link>Interop\Unix\libc\Interop.getsockopt.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.ioctl.cs">
+      <Link>Interop\Unix\libc\Interop.ioctl.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.iovec.cs">
+      <Link>Interop\Unix\libc\Interop.iovec.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.linger.cs">
+      <Link>Interop\Unix\libc\Interop.linger.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.listen.cs">
+      <Link>Interop\Unix\libc\Interop.listen.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.msghdr.cs">
+      <Link>Interop\Unix\libc\Interop.msghdr.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.recv.cs">
+      <Link>Interop\Unix\libc\Interop.recv.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.recvfrom.cs">
+      <Link>Interop\Unix\libc\Interop.recvfrom.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.recvmsg.cs">
+      <Link>Interop\Unix\libc\Interop.recvmsg.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.select.cs">
+      <Link>Interop\Unix\libc\Interop.select.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.send.cs">
+      <Link>Interop\Unix\libc\Interop.send.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.sendmsg.cs">
+      <Link>Interop\Unix\libc\Interop.sendmsg.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.sendto.cs">
+      <Link>Interop\Unix\libc\Interop.sendto.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.setsockopt.cs">
+      <Link>Interop\Unix\libc\Interop.setsockopt.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.shutdown.cs">
+      <Link>Interop\Unix\libc\Interop.shutdown.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.socket.cs">
+      <Link>Interop\Unix\libc\Interop.socket.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.timeval.cs">
+      <Link>Interop\Unix\libc\Interop.timeval.cs</Link>
+    </Compile>
+
+    <Compile Include="$(CommonPath)\Interop\Unix\System.Native\Interop.Close.cs">
+      <Link>Interop\Unix\System.Native\Interop.Close.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\System.Native\Interop.Fcntl.cs">
+      <Link>Interop\Unix\System.Native\Interop.Fcntl.cs</Link>
+    </Compile>
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetsLinux)' == 'true' ">
+    <Compile Include="System\Net\Sockets\SocketAsyncContext.cs" />
+    <Compile Include="System\Net\Sockets\SocketAsyncEngine.cs" />
```

Comment:
They can  be marked as Linux, yes. The picture will get a bit more complicated when OS X enters the picture, however: I expect to reuse large parts of these two types (only the event notification mechanism should be different between the two OSes), so these will probably become SocketAsync{Context,Engine}.{Unix,Linux,OSX}.cs over time.